packages/lines/src/line/Line.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  BufferBindFlag,
+  BufferMesh,
+  BufferUsage,
+  Color,
+  MeshRenderer,
+  Script,
+  VertexElement,
+  VertexElementFormat,
+  Buffer,
+  IndexFormat
+} from "oasis-engine";
+import { LineMaterial } from "./material/LineMaterial";
+import { LineCap, LineJoin } from "./constants";
+import lineBuilder from "./vertexBuilder";
+
+type Point = {
+  x: number;
+  y: number;
+};
+
+export default class Line extends Script {
+  protected _points: Point[] = [];
+  protected _cap = LineCap.Butt;
+  protected _join = LineJoin.Miter;
+  protected _renderer: MeshRenderer;
+  protected _material: LineMaterial;
+  protected _mesh: BufferMesh;
+  protected _needUpdate = false;
+  protected _flattenPoints: number[] = [];
+
+  set points(value: Point[]) {
+    this._points = value;
+    this._flattenPoints = this._points
+      .map((point) => {
+        return [point.x, point.y];
+      })
+      .flat();
+    this._needUpdate = true;
+  }
+
+  set cap(value: LineCap) {
+    if (value !== this._cap) {
+      this._cap = value;
+      this._material.cap = value;
+      this._needUpdate = true;
+    }
+  }
+
+  set join(value: LineJoin) {
+    if (value !== this._join) {
+      this._join = value;
+      this._material.join = value;
+      this._needUpdate = true;
+    }
+  }
+
+  set width(value) {
+    this._material.width = value;
+  }
+
+  set color(value) {
+    this._material.color = value;
+  }
+
+  constructor(entity) {
+    super(entity);
+  }
+
+  protected async generateData() {
+    return await lineBuilder.solidLine(this._flattenPoints, this._join, this._cap, -1);
+  }
+
+  protected async render() {
```

Comment:
For project medthod should use `_render()` instead `render()`,only public method don't need to add `_`

Reply:
fixed